### ISSUE TITLE
Cannot find prebuilt Frida extension

### ISSUE THREAD

**Issue by manuporto** [frida/frida#127]
Hi, I'm trying to install Frida with pip and I get the following error although I'm using Python 2.7
```
Installing collected packages: frida
  Running setup.py install for frida ... error
    Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-NxpB_k/frida/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Z4fUKa-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-i686-2.7
    creating build/lib.linux-i686-2.7/frida
    copying frida/ps.py -> build/lib.linux-i686-2.7/frida
    copying frida/repl.py -> build/lib.linux-i686-2.7/frida
    copying frida/discoverer.py -> build/lib.linux-i686-2.7/frida
    copying frida/tracer.py -> build/lib.linux-i686-2.7/frida
    copying frida/core.py -> build/lib.linux-i686-2.7/frida
    copying frida/application.py -> build/lib.linux-i686-2.7/frida
    copying frida/lsd.py -> build/lib.linux-i686-2.7/frida
    copying frida/__init__.py -> build/lib.linux-i686-2.7/frida
    running build_ext
    downloading prebuilt extension from https://pypi.python.org/packages/2.7/f/frida/frida-7.0.10-py2.7-linux-i686.egg
    trying to get prebuilt extension from the current folder /tmp/pip-build-NxpB_k/frida/frida-7.0.10-py2.7-linux-i686.egg
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-NxpB_k/frida/setup.py", line 138, in <module>
        'build_ext': FridaPrebuiltExt
      File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/local/lib/python2.7/dist-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib/python2.7/distutils/command/install.py", line 601, in run
        self.run_command('build')
      File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/lib/python2.7/distutils/command/build.py", line 128, in run
        self.run_command(cmd_name)
      File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/local/lib/python2.7/dist-packages/setuptools/command/build_ext.py", line 49, in run
        _build_ext.run(self)
      File "/usr/lib/python2.7/distutils/command/build_ext.py", line 337, in run
        self.build_extensions()
      File "/usr/lib/python2.7/distutils/command/build_ext.py", line 446, in build_extensions
        self.build_extension(ext)
      File "/tmp/pip-build-NxpB_k/frida/setup.py", line 80, in build_extension
        raise Exception("Could not find prebuilt Frida extension. "
    Exception: Could not find prebuilt Frida extension. Prebuilts only provided for python 2.6-2.7 and 3.5
    
    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-NxpB_k/frida/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Z4fUKa-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-NxpB_k/frida/
```

* Ubuntu 14.04
* Python 2.7.6
* Pip 8.1.1

**Comment by oleavr** [comment]
Hi,

It looks like it's trying to download the following prebuild:
`https://pypi.python.org/packages/2.7/f/frida/frida-7.0.10-py2.7-linux-i686.egg`
Which does indeed exist, so this looks like a network / firewall issue. Could you try downloading [the latest source tarball](https://pypi.python.org/packages/source/f/frida/frida-7.0.10.zip#md5=b1cb406f104a053ba20d54f0b08c8dac) and adding some temporary logging to `setup.py` to see what's causing the error?

Cheers!

**Comment by manuporto** [comment]
Hi,

Sorry for the delay. I added some `print`'s so I could have more information about the Exception's but I'm not getting anything useful. Here is the edited code:

```python
try:
    print('downloading prebuilt extension from', egg_url)
    egg_data = urlopen(egg_url).read()
except Exception as e:
    print("Error {0}: {1}".format(e.errno, e.strerror)) #Here
    egg_path = os.path.join(os.getcwd(), egg_filename)
    print('trying to get prebuilt extension from the current folder', egg_path)
``` 

And this it's what I'm getting:
```
running build
running build_py
running build_ext
downloading prebuilt extension from https://pypi.python.org/packages/3.4/f/frida/frida-7.0.10-py3.4-linux-i686.egg
Error None: None <---------------------------------------- This is the print
trying to get prebuilt extension from the current folder /home/mporto/Downloads/frida-7.0.10/frida-7.0.10-py3.4-linux-i686.egg
Error 2: No such file or directory
Traceback (most recent call last):
  File "setup.py", line 140, in <module>
    'build_ext': FridaPrebuiltExt
  File "/usr/lib/python3.4/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.4/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.4/distutils/command/build.py", line 126, in run
    self.run_command(cmd_name)
  File "/usr/lib/python3.4/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/lib/python3/dist-packages/setuptools/command/build_ext.py", line 49, in run
    _build_ext.run(self)
  File "/usr/lib/python3.4/distutils/command/build_ext.py", line 339, in run
    self.build_extensions()
  File "/usr/lib/python3.4/distutils/command/build_ext.py", line 448, in build_extensions
    self.build_extension(ext)
  File "setup.py", line 82, in build_extension
    raise Exception("Could not find prebuilt Frida extension. "
Exception: Could not find prebuilt Frida extension. Prebuilts only provided for python 2.6-2.7 and 3.5
```

I don't know of other way of logging what happens with urlopen, so if you do I will be glad to try it out.

Cheers!

**Comment by manuporto** [comment]
To anyone that happens to have this problem, now or in the future. A workaround to this problem is to download the [latest source tarball](https://pypi.python.org/packages/source/f/frida/frida-7.0.10.zip#md5=b1cb406f104a053ba20d54f0b08c8dac) and the corresponding `.egg` [file](https://pypi.python.org/pypi/frida) and run `setup.py`. Thanks to the Frida developers, this option is available and you can download and install Frida behind a proxy.

**Comment by pentagramz** [comment]
![frida1](https://cloud.githubusercontent.com/assets/7072014/14763315/8ea95216-09af-11e6-9c6f-2baf590c048b.JPG)

I downloaded the latest version tarball and corresponding .egg file .Not able to run setup.py?
Please support

**Comment by manuporto** [comment]
@pentagramz you are not using `setup.py` correctly, see it's usage (it's printed in your cmd).

**Comment by pentagramz** [comment]
Hello I tried python setup.py install and this is the error I am facing now.
Any pointers?
![frida3](https://cloud.githubusercontent.com/assets/7072014/14766147/bf385fd6-0a1c-11e6-816b-86d9516f6ae7.JPG)
![frida4](https://cloud.githubusercontent.com/assets/7072014/14766151/c7465b1a-0a1c-11e6-8837-cc0dfbea7714.JPG)

**Comment by gi0cann** [comment]
I may be wrong, but it looks like you have a .egg for python 3.5 and you are using python 2.7 to run setup .py. Run python -V to make sure you have the correct version of python.

**Comment by mike4420** [comment]
There is no linux egg available for the latest version of Frida (7.2.4).
https://pypi.python.org/pypi/frida

**Comment by oleavr** [comment]
@mike4420 This should be fixed now, was a build machine hiccup.
